src/crypto/tls/ech_config.go

```diff
@@ -87,13 +88,12 @@ func echMarshalConfigs(configs []ECHConfig) ([]byte, error) {
 
 func readConfigContents(contents *cryptobyte.String, config *ECHConfig) bool {
 	var t cryptobyte.String
-	if !contents.ReadUint16LengthPrefixed(&t) ||
-		!t.ReadBytes(&config.rawPublicName, len(t)) ||
-		!contents.ReadUint16LengthPrefixed(&t) ||
-		!t.ReadBytes(&config.rawPublicKey, len(t)) ||
+	if !contents.ReadUint8(&config.configId) ||
 		!contents.ReadUint16(&config.kemId) ||
 		!contents.ReadUint16LengthPrefixed(&t) ||
-		len(t)%4 != 0 {
+		!t.ReadBytes(&config.rawPublicKey, len(t)) ||
+		!contents.ReadUint16LengthPrefixed(&t) {
+		fmt.Println("here")
```

Comment:
Issues:
- Excess debug print
- Missing check that `cipher_suites` is actually a multiple of 4 (the `len(t)%4 != 0` check needs to be added to the end).
- Consider adding a comment here that `HpkeKeyConfig` is being parsed here (`cipher_suites` will be parsed below though).

Reply:
Hah oops, missed this one.

Reply:
I didn't add clarifying comments since the intent seems clear as-is. Addressed the rest, though!